network.go

```diff
@@ -736,6 +736,15 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 		ep.ipamOptions[netlabel.MacAddress] = ep.iface.mac.String()
 	}
 
+	if ipam.capability.RequiresEndpointInfo {
+		if ep.ipamOptions == nil {
+			ep.ipamOptions = make(map[string]string)
+		}
+		log.Debugf("Endpoint: %s", ep)
+		ep.ipamOptions[netlabel.EndpointName] = name
```

Comment:
I am not sure we are ok in providing the remote IPAM drivers with the endpoint name and network name (change below).

Please take a look at @mrjana explanation in https://github.com/docker/libnetwork/issues/139#issuecomment-110066534